freetype/__init__.py

```diff
@@ -977,17 +977,25 @@ def __init__( self, filename, index = 0 ):
         self._FT_Face = None
         #error = FT_New_Face( library, filename, 0, byref(face) )
         self._filebodys = []
-        try:
-            u_filename = c_char_p(_encode_filename(filename))
-            error = FT_New_Face( library, u_filename, index, byref(face) )
-        except UnicodeError:
-            with open(filename, mode='rb') as f:
-                filebody = f.read()
+        if hasattr(path_or_file, 'read'):
+            filebody = path_or_file.read()
             error = FT_New_Memory_Face( library, filebody, len(filebody),
                                         index, byref(face) )
             self._filebodys.append(filebody)  # prevent gc
+            self._filename = ":memory:"
```

Comment:
Could probably just make it None here

Reply:
sqlite3 uses `:memory:` to denote a memory file, I thought this make sense.

Reply:
Using `None`, you could in code later use `if not x.filename: save or something`. Although I'm not a fan of assigning `None` to anything other than using if for default arguments, so assigning an empty string `""` would be even nicer. Unless the code at some point wants to switch to using pathlib.Path objects for all path related things.